src/eda.py

```diff
@@ -0,0 +1,202 @@
+    font_size : int, optional
+        [font size], by default 14
+    header_color : str, optional
+        [header color], by default '#40466e'
+    row_colors : list, optional
+        [row color], by default ['#f1f1f2', 'w']
+    edge_color : str, optional
+        [edge color], by default 'w'
+    bbox : list, optional
+        [bbox ], by default [0, 0, 1, 1]
+    header_columns : int, optional
+        [header columns], by default 0
+    ax : [type], optional
+        [plotting table, by default None
+
+    Returns
+    -------
+    [object]
+        [figure]
+    """
+    if ax is None:
+        size = (np.array(data.shape[::-1]) + np.array([0, 1])) * np.array([col_width, row_height])
+        fig, ax = plt.subplots(figsize=size)
+        ax.axis('off')
+    mpl_table = ax.table(cellText=data.values, bbox=bbox, colLabels=data.columns, **kwargs)
+    mpl_table.auto_set_font_size(False)
+    mpl_table.set_fontsize(font_size)
+
+    for k, cell in mpl_table._cells.items():
+        cell.set_edgecolor(edge_color)
+        if k[0] == 0 or k[1] < header_columns:
+            cell.set_text_props(weight='bold', color='w')
+            cell.set_facecolor(header_color)
+        else:
+            cell.set_facecolor(row_colors[k[0]%len(row_colors) ])
+    return ax.get_figure(), ax
+
+def generate_dataset_overview(data_frame, output_folder, file_name):
+    """
+    Generates an overview of the dataset.
+    Also saves resulting table as file in given output folder.
+    Parameters:
+    -----------
+    data_frame : pandas.DataFrame
+        input path to be verified
+    output_folder : str
+        output folder path to save the chart
+    file_name : str
+        file name for generated chart image
+        
+    Returns:
+    -----------
+    str
+        saved file path
+    """
+    data_overview = [
+            {"Dataset": "Number of features", "Value": len(data_frame.columns)},
+            {"Dataset": "Number of characters", "Value": len(data_frame)},
+            {"Dataset": "Number of Missing cells", "Value": (data_frame.isnull()).sum().sum()},
+            {"Dataset": "Percentage of Missing cells", "Value": round((data_frame.isnull()).sum().sum()/data_frame.size*100, 2)}
+        ]
+    overview_frame = pd.DataFrame(data_overview)
+    fig_1, ax_1 = render_mpl_table(overview_frame, header_columns=0, col_width=5)
+
+    return fig_1.savefig(output_folder +"/"+ file_name)
+def generate_feature_overview(data_frame, output_folder, file_name):
+    """
+    Generates an overview of the features in dataset.
+    Also saves resulting table as file in given output folder.
+    Parameters:
+    -----------
+    data_frame : pandas.DataFrame
+        input path to be verified
+    output_folder : str
+        output folder path to save the chart
+    file_name : str
+        file name for generated chart image
+        
+    Returns:
+    -----------
+    object
+        saved .png file
+    """
+    dict2 = dict()
```

Comment:
Consistent spacing, and more explanatory variable names. What are dict2 and dict3 referring to?